webapp/src/ts/modules/tasks/tasks.component.ts

```diff
@@ -118,50 +128,91 @@ export class TasksComponent implements OnInit, OnDestroy {
 
   private hydrateEmissions(taskDocs) {
     return taskDocs.map(taskDoc => {
-      const emission = { ...taskDoc.emission };
+      const emission = {...taskDoc.emission};
       const dueDate = moment(emission.dueDate, 'YYYY-MM-DD');
       emission.date = new Date(dueDate.valueOf());
       emission.overdue = dueDate.isBefore(moment());
       emission.owner = taskDoc.owner;
-
+      emission.forId = taskDoc.forId ? taskDoc.forId : taskDoc.owner;
       return emission;
     });
   }
 
-  private refreshTasks() {
-    const telemetryData:any = {
-      start: Date.now(),
-    };
-
-    return this.rulesEngineService
-      .isEnabled()
-      .then(isEnabled => {
-        this.tasksDisabled = !isEnabled;
-        return isEnabled ? this.rulesEngineService.fetchTaskDocsForAllContacts() : [];
-      })
-      .then(taskDocs => {
-        this.hasTasks = taskDocs.length > 0;
-        this.loading = false;
-        this.tasksActions.setTasksList(this.hydrateEmissions(taskDocs));
-        if (!this.tasksLoaded) {
-          this.tasksActions.setTasksLoaded(true);
+  private async refreshTasks() {
+    try {
+      const telemetryData: any = {
+        start: Date.now(),
+      };
+
+      const isEnabled = await this.rulesEngineService.isEnabled();
+      this.tasksDisabled = !isEnabled;
+      const taskDocs = isEnabled ? await this.rulesEngineService.fetchTaskDocsForAllContacts() : [];
+
+      this.hasTasks = taskDocs.length > 0;
+      this.loading = false;
+
+      const hydratedTasks = await this.hydrateEmissions(taskDocs) || [];
+      const subjects = await this.getLineagesFromTaskDocs(hydratedTasks);
+      if (subjects?.size) {
+        for (const task of hydratedTasks) {
```

Comment:
In theory the `this.currentLevel` promise should resolve immediately the following iterations and the performance impact of awaiting for every task shouldn't be big.... 

Another option is to await 1 time first and send over the currentLevel to the `getTaskLineage`, something like this:
```
... other code...

const subjects = await this.getLineagesFromTaskDocs(hydratedTasks);

if (subjects?.size) {
  const userLineageLevel = await this.currentLevel;
  hydratedTasks.forEach(task => task.lineage = this.getTaskLineage(subjects, task, userLineageLevel));
}
...continue code...
``` 
And the `cleanAndRemoveCurrentLineage` wont' be `async` anymore.

@jkuester @elvisdorkenoo what do you think?

Reply:
Yeah I like the second option, it would make me passing over `userLineageLevel `  parameter, but it's fine.